orangecontrib/spectroscopy/widgets/owpreprocess.py

```diff
@@ -1780,7 +1781,7 @@ def set_data(self, data=None):
 
     def handleNewSignals(self):
         self.show_preview(True)
-        self.apply()
+        self.unconditional_commit()
```

Comment:
Do we really want to by-pass the auto-commit logic when we connect new data? Users expect if that box is unchecked that no calculations take place (important for large datasets).

Reply:
It is how (other) Orange widgets currently work. We try to encourage users that want to prevent propagation at new data by pausing a connection (right click it and you have an option for that).

Until recently, I though that this works differently, but last time I was playing with loading -> preprocess -> ft (or reverse) -> hyperspectra and was happy that it works such as is. But I do not know which would be better really... The current implementation is good if you want to use a workflow to compare effects on different inputs, while what you propose could be better if you want user to modify prebuilt workflows for their data.

Reply:
I'll admit I'm a bit confused: I thought it worked like this:

1. Auto-commit on: any change to data input / widget settings is immediately propagated to the outputs
2. Auto-commit off: any change to data input is reflected in widget previewing, but the heavy-lifting / full-data calculations (i.e. Preprocessing, clustering, FFT, etc) is postponed until the user presses the "Commit" button.

I think 2. is superior to freezing the connection because you can still see the preview/info boxes. It is inferior because you can't tell from the canvas which widgets are not calculating automatically.

> The current implementation is good if you want to use a workflow to compare effects on different inputs, while what you propose could be better if you want user to modify prebuilt workflows for their data.

Hopefully we can find a way to support both? I thought this is what auto-commit check box is for.

Reply:
What you are saying makes sense. This widget really is quite special because seeing a preview before doing something compute heavy is a must.

Did it ever work as your are saying and I recently changed it for the worse?

Reply:
I don't recall if this particular widget worked that way :) I will do some investigating and open a new issue if necessary to track it.